projects/packages/autoloader/src/class-plugin-locator.php

```diff
@@ -110,6 +116,67 @@ public function find_using_request_action( $allowed_actions ) {
 		return $this->convert_plugins_to_paths( $plugin_slugs );
 	}
 
+	/**
+	 * Inspects the request to determine if this is a Woo plugin actiation request. If it is,
+	 * returns the list of activating plugin slugs.
+	 *
+	 * @return array|false Returns an array of plugin slugs or false if this is not a Woo plugin
+	 *                     activation request.
+	 */
+	private function find_activating_plugins_in_woo_request() {
+		// Check for a Woo plugin activation request.
+		if ( ! isset( $_SERVER['REQUEST_URI'] ) || false === strpos( $_SERVER['REQUEST_URI'], 'wc-admin/plugins/activate' ) ) {
+			return false;
+		}
+
+		$input              = json_decode( file_get_contents( 'php://input' ) );
+		$activating_plugins = isset( $input->plugins ) ? $input->plugins : null;
+
+		if ( ! isset( $activating_plugins ) ) {
+			return false;
+		}
+
+		$activating_plugins = explode( ',', $activating_plugins );
+
+		if ( ! is_array( $activating_plugins ) ) {
+			return false;
+		}
+
+		// These plugins are from the list in Automattic\WooCommerce\Admin\Features\Onboarding::get_onboarding_allowed_plugins.
+		$allowed_plugins = array(
```

Comment:
Given the [volatility of this list](https://github.com/woocommerce/woocommerce-admin/commits/main/src/Features/Onboarding.php), I wonder if we should just skip the validation. It looks like `$plugin/$plugin.php` would work in all of the given cases, and the `convert_plugins_to_paths()` only allows for slugs in a plugin directory.